giraffa-core/src/main/java/org/apache/giraffa/hbase/NamespaceProcessor.java

```diff
@@ -942,6 +943,22 @@ public void rename2(String src, String dst, Rename... options)
       rootDstNode = null;
     }
 
+    /*
+     * If src and dst key bytes are equal, we can simply update the path since
+     * no new row needs to be created. This will occur during in-place renames
+     * using id-based row keys.
+     */
+    if (rootSrcNode != null) {
+      RowKey rootSrcKey = rootSrcNode.getRowKey();
+      RowKey rootDstKey = keyFactory.newInstance(dst, rootSrcNode.getId());
+      if (Arrays.equals(rootSrcKey.getKey(), rootDstKey.getKey())) {
+        LOG.debug("In-place rename of " + src + " to " + dst);
+        rootDstNode = rootSrcNode.cloneWithNewRowKey(rootDstKey);
```

Comment:
By "simply update the path" I mean we don't need to create a new row in hbase since the row key bytes stay the same. We still have to create a new INode object with a new RowKey since the RowKey is tied to particular path. To avoid that, we would have to make the path be a mutable field of INode rather than than of RowKey.